src/pip/_internal/index.py

```diff
@@ -63,6 +63,11 @@
     CandidateSortingKey = (
         Tuple[int, int, int, _BaseVersion, BuildTag, Optional[int]]
     )
+    # The CollectedLinks triple consists of--
+    # (1) links from file locations,
+    # (2) links from find_links, and
+    # (3) a dict mapping HTML page url to links from that page.
+    CollectedLinks = Tuple[List[Link], List[Link], Dict[str, List[Link]]]
```

Comment:
A good follow up might be to convert this to a TypedDict, so that this value has some hint in the key names, about what the different values are for.

Reply:
Or a namedtuple, you mean?

Reply:
+1 for namedtuple (and I prefer it over typeddict)